utils/utils.py

```diff
@@ -78,6 +78,15 @@ def make_env(env_id, rank=0, seed=0, log_dir=None):
     def _init():
         set_global_seeds(seed + rank)
         env = gym.make(env_id)
+
+        # from baselines
+        #flatten_dict_observations = alg not in {'her'}
+        flatten_dict_observations = True
+        # check if it is a MiniGrid observation space
+        if flatten_dict_observations and isinstance(env.observation_space, gym.spaces.Dict):
+            keys = env.observation_space.spaces.keys()
+            env = gym.wrappers.FlattenDictWrapper(env, dict_keys=list(keys))
```

Comment:
I did not know about that wrapper..., so from what I understand, it transform everything to a 1D vector?
In that case, the image is not normalized anymore...

Another solution, the goal env wrapper that will released where HER is ready: https://github.com/hill-a/stable-baselines/pull/273

Reply:
Right.  I intended to provide a worst case fallback.  Will check on the other one.

Reply:
I have just read through the thread.  Right now it requires "all keys to have the same type", which is not the case, as it is unfortunately.  Need to look into it further to understand its ideas.